### PR TITLE
Support TRAMP

### PR DESCRIPTION
**Pull request by Silex**
Hello!

With very little modification emacs-request now supports TRAMP! This is useful when you want your requests to be run on the remote host you're currently on, which is my case.

The first hunk could probably be made prettier, 99% of the trick is simply calling `start-file-process` instead of `start-process`... but I had to modify the first hunk because it explicitely changed default-directory to `~/`, effectively forcing it to execute on the local machine instead of the remote one.

If the default-directory manipulation is not that important, we could envisage to completely remove it. It was introduced by 584338027cfd6033ba95330483e646afeea75dfa.

### CODE REVIEW

**Comment by tzz** [comment]
whoa this is nice :)

File: request.el [hunk 1]
```diff
@@ -977,15 +977,19 @@ removed from the buffer before it is shown to the parser function.
   (let* (;; Use pipe instead of pty.  Otherwise, curl process hangs.
          (process-connection-type nil)
          ;; Avoid starting program in non-existing directory.
-         (default-directory (expand-file-name "~/"))
+         (home-directory (if (tramp-tramp-file-p default-directory)
```
Comment (syohex) [inline request.el]: Can `file-remote-p` be used instead `tramp-tramp-file-p` ? `tramp.el` is loaded when `tramp-tramp-file-p` is called. tramp.el is a big library, non-tramp users should not load it if it is needed.

Comment (Silex) [inline request.el]: Good idea, let me check.

**Comment by Silex** [comment]
Done! Seems to work fine.

**Comment by syohex** [comment]
Thanks

**Comment by Silex** [comment]
:+1:

**Comment by marsam** [comment]
Just a little note, this change doesn't mean full tramp support, when `:data` parameter is passed request.el pass it to curl from [stdin](https://github.com/tkf/emacs-request/blob/a65e11a/request.el#L898) so when `default-directory` is a remote directory will left a curl process hanging, unless request-timeout is set.
I made a few changes to request.el to support tramp: https://github.com/emacs-pe/emacs-request/commit/7d1cb90, I'll make a pull request later.

**Comment by Silex** [comment]
Ah, sorry I missed it :worried:

**Comment by Silex** [comment]
@marsam: did you do the PR for this?

**Pull request merged**